backend/stakepoold/userdata/userdata.go

```diff
@@ -0,0 +1,126 @@
+package userdata
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"sync"
+)
+
+type DBConfig struct {
+	DBHost     string
+	DBName     string
+	DBPassword string
+	DBPort     string
+	DBUser     string
+}
+
+// UserData stores the current snapshot of the user voting config.
+type UserData struct {
```

Comment:
This lock is guaranteed wrong.  This needs to be in appcontext because we need to serialize pointer access; not the data fields themselves.